polkadot/node/network/approval-distribution/src/lib.rs

```diff
@@ -1305,6 +1423,9 @@ impl State {
 				continue
 			}
 
+			if !topology.map(|topology| topology.is_validator(&peer)).unwrap_or(false) {
```

Comment:
This seems unrelated. Where is this coming from?

Reply:
It was discovered, during my testing, that we are sending the random assignments to non-validators as well, which is wrong because those will not forward it because they do not have a grid topology, so we were wasting random assignments when sending them that way.


But, I agree it doesn't have to be part of this PR, I will split in a separate PR.

Reply:
No need. Was just wondering.

Reply:
Stating in the description for reference that this was fixed is enough.